### DISCUSSION TITLE
Cache-control header on 500 error page response

### DISCUSSION THREAD

**Discussion by dejavu1987** [vercel/next.js#49150] [Help]
### Summary

It seems that nextjs sets `cache-control` header to `s-maxage=31536000, stale-while-revalidate`
for a 500 response.

The issue is we are using Cloudfront CDN in front of the next server so whenever there is a 500 error, the page got cached for ever.

I have tried setting the header to something else in `getServerSideProps`, but it seems to be overriden by next, I could add any other arbitrary header but could not change the `cache-control`

Any help will be appreciated.

Regards

**Comment by dejavu1987** [comment]
I realised `getStaticProps` allowed to add revalidate to change the `s-maxage` on the response header but we are looking at adding `Cache-Control: private, no-store` to the header so that the 500 error is not cached on CDN.
Is it possible to do so?

**Comment by rodrigo-arias** [comment]
Just to mention that we are experiencing the same thing. Sometimes our servers can't respond and we get 500 errors. Even if the server can respond 5 seconds later, these pages remain cached and the site needs to be rebuilt. Hosted on WP Engine which uses Cloudflare.

**Comment by rodrigo-arias** [comment]
@timneutkens is it possible to replicate the solution that you added https://github.com/vercel/next.js/pull/24983 according to the comment https://github.com/vercel/next.js/issues/22579#issuecomment-838346238 for page 500? Thank you.

**Comment by evilj0e** [comment]
We've faced the same issue and it seems that the best solution will be to set `Cache-Control` header manually.
I hope it helps.

```
// For instance for api route located in the /pages/api/some-route.ts
export default async (req: NextApiRequest, res: NextApiResponse) => {
   res.setHeader('Cache-Control', 'no-store')
   
   // do something useful here
}
```
